app/models/report.rb

```diff
@@ -60,46 +60,190 @@ def card_overall(collection, term)
+    Thesis.all.joins(:copyright).group(:holder).group(:grad_date).count.each do |item|
+      rows[item[0][0]][item[0][1]] = item[1]
+    end
+    rows.each do |row|
+      category.push({
+                      label: row[0],
+                      data: pad_row(row[1])
+                    })
+    end
+    category
   end
 
-  def dashboard_tables(collection)
-    result = {}
-    result['status'] = table_status collection
-    result['copyright'] = table_copyright collection
-    result
+  def data_category_department
+    category = []
+    rows = pad_category(Department.pluck(:name_dw))
+    Thesis.all.joins(:departments).group(:name_dw).group(:grad_date).count.each do |item|
+      rows[item[0][0]][item[0][1]] = item[1]
+    end
+    rows.each do |row|
+      category.push({
+                      label: row[0],
+                      data: pad_row(row[1])
+                    })
+    end
+    category
   end
 
-  def extract_terms(collection)
-    collection.pluck(:grad_date).uniq.sort
+  def data_category_license
+    category = []
+    rows = pad_category(License.pluck(:display_description))
+    Thesis.all.joins(:license).group(:display_description).group(:grad_date).count.each do |item|
+      rows[item[0][0]][item[0][1]] = item[1]
+    end
+    rows.each do |row|
+      category.push({
+                      label: row[0],
+                      data: pad_row(row[1])
+                    })
+    end
+    category
   end
 
-  def table_populate_defaults(data, values)
-    values.each do |value|
-      data[value] = '0' unless data[value]
+  def data_category_publication_status
+    category = []
+    rows = pad_category(Thesis.publication_statuses)
+    Thesis.all.group(:publication_status).group(:grad_date).count.each do |item|
+      rows[item[0][0]][item[0][1]] = item[1]
+    end
+    rows.each do |row|
+      category.push({
+                      label: row[0],
+                      data: pad_row(row[1])
+                    })
     end
+    category
   end
 
-  def table_status(collection)
-    result = collection.group(:publication_status).count
-    table_populate_defaults result, Thesis.publication_statuses
+  def data_files_attached_to_theses
     {
-      'title' => 'Thesis counts by publication status',
-      'summary' => 'This table presents a summary of thesis records by their publication status. The second column '\
-                   'gives the status, while the first column gives how many records have that status.',
-      'column' => 'Publication status',
-      'data' => result
+      label: 'Files attached to theses',
+      data: pad_row(Thesis.all.joins(:files_attachments).group(:grad_date).count)
     }
   end
 
+  def data_issues
+    {
+      label: 'Flagged with issues',
+      data: pad_row(Thesis.all.group(:grad_date).where('issues_found = ?', true).count)
+    }
+  end
+
+  def data_multiple_authors
+    row_data = {}
+    sql = 'SELECT t.grad_date, count(t.id) as pop '\
```

Comment:
Did you consider heredoc or %Q syntax for multiline strings?

Reply:
Having looked again at these syntaxes, I think Heredoc makes the most sense. I like its compactness compared with the blank lines needed by %Q (especially given how long these methods are already). Thanks for the suggestion! This will be addressed by the next commit (underway now).